[no-master] Enable Scala 2.13.0 in the tools and all the other artifacts.

sjrd opened the pull request:
It's [no-master]  because

* The `@deprecatedName` and `scopt` commits are irrelevant in master.
* The other two commits are virtually all-conflicts if I try to cherry-pick them on master. An independent forward-port will be easier to review.

gzm0 commented on tools/shared/src/main/scala/org/scalajs/core/tools/linker/CollectionsCompat.scala:
```diff
@@ -0,0 +1,32 @@
+
+    // filterInPlace replaces retain
+    def filterInPlace(p: (K, V) => Boolean): Unit = {
+      // Believe it or not, this is the implementation of `retain` in 2.12.x:
```
Hahahahaha, were you anticipating me complaining :P

sjrd replied:
Actually I was anticipating myself looking back at this code and being appalled at the inefficiency of it.

gzm0 commented on tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/GenIncOptimizer.scala:
```diff
@@ -1027,31 +1029,37 @@ object GenIncOptimizer {
     def emptyParIterable[V]: ParIterable[V]
 
     // Operations on ParMap
+    def parMapIsEmpty[K, V](map: ParMap[K, V]): Boolean
```
Simply name this `isEmpty`? `parMap` seems superfluous (I understand that overloading is a problem with types that end up being erased, but the whole thing is already terribly inconsistent anyways :S).

sjrd commented:
Updated.

Besides addressing the review, I have also added the whitelist and blacklist for the `scalaTestSuite`, so that the CI can pass. This discovered two bugs filed as #3704 and #3705, which should be fixed separately, as followups.

gzm0 commented on scala-test-suite/src/test/resources/2.13.0/BlacklistedTests.txt:
```diff
@@ -122,12 +172,20 @@ scala/util/matching/RegexTest.scala
-
-# Bugs
-scala/collection/convert/MapWrapperTest.scala
+# Fails for a BigDecimal range with augmented precision (might be an actual bug)
```
Do we need to file an issue for this at least?

sjrd replied:
Filed as #3706.
